test/widgets/content_test.dart

```diff
@@ -290,6 +297,79 @@ void main() {
 
   testContentSmoke(ContentExample.mathBlock);
 
+  testContentSmoke(ContentExample.strong);
+
+  testContentSmoke(ContentExample.emphasis);
+
+  group('InlineCodeNode', () {
+    testContentSmoke(ContentExample.inlineCode);
+
+    testWidgets('maintains font-size ratio with surrounding text', (tester) async {
+      /// Simulate a nested span's style by merging all ancestor-span styles,
+      /// starting from the root.
+      ///
+      /// This isn't how the style actually gets applied in the code under test,
+      /// but it should hopefully simulate it closely enough.
+      TextStyle? mergeStylesOuterToInner(InlineSpan outerSpan, InlineSpan innerSpan) {
+        final styles = <TextStyle?>[];
+        bool recurse(InlineSpan span) {
+          styles.add(span.style);
+          if (span == innerSpan) {
+            return false;
+          }
+          final notInSubtree = span.visitDirectChildren(recurse);
+          if (notInSubtree) {
+            styles.removeLast();
+          }
+          return notInSubtree;
+        }
+        recurse(outerSpan);
+
+        return styles.reduce((value, element) => switch ((value, element)) {
+          (TextStyle(), TextStyle()) => value!.merge(element!),
+          (TextStyle(), null)        => value,
+          (null, TextStyle())        => element,
+          (null, null)               => null,
+        });
+      }
+
+      TextStyle? mergedStyleOfSubstring(InlineSpan rootSpan, String substring) {
+        final substringSpan = rootSpan.getSpanForPosition(
+          TextPosition(offset: rootSpan.toPlainText().indexOf(substring)));
+        check(substringSpan).isNotNull();
+        return mergeStylesOuterToInner(rootSpan, substringSpan!);
```

Comment:
This works fine for this test, and if you find it works fine for the other test cases then it can stay just as it is.

A potentially cleaner, and certainly more flexible, version would have the recursion take not an InlineSpan to look for, but a predicate to call on each descendant span to see if it's the one we're looking for. Then the predicate here can be like `(span) => span.text == substring`. That way we recurse through the tree only once, and it opens the way for different predicates for other test cases if needed.